routes/definitions.js

```diff
@@ -95,15 +95,32 @@ async function reload(request, response) {
 // Typically used by a UI to preview the effect of a patch
 router.post(
   '/:type/:provider/:namespace/:name/:revision',
-  asyncMiddleware(async (request, response) => {
-    if (!request.query.preview)
-      return response.status(400).send('Only valid for previews. Use the "preview" query parameter')
-    if (!validator.validate('curation', request.body)) return response.status(400).send(validator.errorsText())
-    const coordinates = await utils.toEntityCoordinatesFromRequest(request)
-    const result = await definitionService.compute(coordinates, request.body)
-    response.status(200).send(result)
-  })
+  asyncMiddleware(previewWithCoordinatesParams)
 )
+async function previewWithCoordinatesParams(request, response) {
+  if (!request.query.preview)
+    return response.status(400).send('Only valid for previews. Use the "preview" query parameter')
+  await previewDefinition(request, request.body, response)
+}
+
+async function previewDefinition(coordinatesParams, curation, response) {
+  if (!validator.validate('curation', curation)) return response.status(400).send(validator.errorsText())
+  const coordinates = await utils.toEntityCoordinatesFromRequest(coordinatesParams)
+  const result = await definitionService.compute(coordinates, curation)
+  response.status(200).send(result)
+}
+
+// Previews the definition for a component aggregated and with the given curation applied.
+// Typically used by a UI to preview the effect of a patch
+// The request body contains:
+// - coordinates: component coordinates as a string
+// - curation: the curation to be applied
+router.patch('/', asyncMiddleware(previewWithCoordinates))
```

Comment:
Instead of using the patch, can we use post? `patch` is basically for updates. If conflict with other routes, we could rename the route to be `/preview`.

Reply:
I originally did it as a Post call to definitions/preview; however, after reviewing the API [swagger-ui](https://api.clearlydefined.io/api-docs/#), we don't tend to add any path (e.g. /preview) in api endpoints.  In general, Patch is for partial update, and Post is for creating new resources (see https://www.w3schools.com/tags/ref_httpmethods.asp).  That is why I put it as a patch call.

If you prefer to have it as a post call,  I can change it back.

Reply:
As per suggestion, updated to a post call to definitions/preview.  See https://github.com/clearlydefined/service/compare/9d340fd6ca13243a8386b31b9143f6a496a8ef07..81433237d5456b8ee650f6e0104283c3a87580b2

Reply:
Another thought, to align with coordinates as query parameters (as in definitions-1.0.0), can potentially put coordinates as query params: patch /definitions?coordinates=zzzz, request body contains the same curation body currently in the build (prior to change)  What do you think?

Reply:
After reading about REST API designs, the coordinates here are the identifiers for a specific definition resource, and therefore, belongs to the url.  Fix parsing coordinates in the path instead.  See Route paths https://expressjs.com/en/guide/routing.html